pkgs/by-name/bi/bilibili/package.nix

```
@@ -1,26 +1,33 @@
-{ lib
-, stdenv
-, fetchurl
-, electron
-, zstd
-, makeWrapper
+{
+  lib,
+  stdenv,
+  fetchurl,
+  electron,
+  dpkg,
+  makeWrapper,
+  commandLineArgs ? "",
 }:
-
-stdenv.mkDerivation rec {
-  pname = "bilibili";
-  version = "1.14.0-1";
-  src = fetchurl {
-    url = "https://github.com/msojocs/bilibili-linux/releases/download/v${version}/io.github.msojocs.bilibili_${version}_amd64.deb";
-    hash = "sha256-4+DGL/DNR3wLFUff17OquAM6dOkcsXFNeCqA7ITtCaI=";
+let
+  version = "1.14.0-2";
+  srcs = {
+    x86_64-linux = fetchurl {
+      url = "https://github.com/msojocs/bilibili-linux/releases/download/v${version}/io.github.msojocs.bilibili_${version}_amd64.deb";
+      hash = "sha256-QQMdEpKE7r/fPMaX/yEoaa7KjilhiPMYLRvGPkv1jds=";
+    };
+    aarch64-linux = fetchurl {
+      url = "https://github.com/msojocs/bilibili-linux/releases/download/v${version}/io.github.msojocs.bilibili_${version}_amd64.deb";
+      hash = "sha256-UaGI4BLhfoYluZpARsj+I0iEmFXYYNfl4JWhBWOOip0=";
+    };
   };
-
+  src =
+    srcs.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+in
+stdenv.mkDerivation {
+  pname = "bilibili";
+  inherit src;
+  inherit version;
   unpackPhase = ''
-    runHook preUnpack
-
-    ar x $src
-    tar -I ${zstd}/bin/zstd -xvf data.tar.zst
-
-    runHook postUnpack
+    ${dpkg}/bin/dpkg -x $src ./
```

```suggestion
    runHook preUnpack
    
    ${dpkg}/bin/dpkg -x $src ./
    
    runHook postUnpack
```

Minor thing, but it's a good idea to add the `pre`/`post` hooks when overriding phases.